### PR TITLE
Using hook_modules_installed instead of hook_install to force rest co…

### PR DESCRIPTION
**Pull request by dannylamb**
…nfig

**GitHub Issue**: Islandora-CLAW/CLAW#643

# What does this Pull Request do?

Moves forcing `rest` config for nodes from `hook_install` to `hook_modules_installed`.

# How should this be tested?

A happy Travis is all we need here.

# Interested parties
@Islandora-CLAW/committers

### CODE REVIEW

File: islandora.module [hunk 1]
```diff
@@ -27,14 +28,33 @@ function islandora_help($route_name, RouteMatchInterface $route_match) {
...
     default:
   }
 }
 
 /**
+ * Implements hook_modules_installed().
+ */
+function islandora_modules_installed($modules) {
+  // Ensure the auth and serialization formats we need are available.
+  if (in_array('rest', $modules) {
```
Comment (whikloj) [inline islandora.module]: missing `)`

**Comment by dannylamb** [comment]
@whikloj derp

**Comment by dannylamb** [comment]
Just realizing now that there's two cases we need to handle: when `rest` is already enabled and when `rest` is enabled along with islandora.

In its current state, Travis fulfills the "rest already enabled" case, so let's make Travis green with that.  Then we'll revert https://github.com/Islandora-CLAW/CLAW/pull/652 and make sure it's green there too.

_Then_ we can resolve Islandora-CLAW/CLAW#643.

**Comment by dannylamb** [comment]
@ruebot You mind merging the revert pull so we can test what happens if you _don't_ have `rest` enabled beforehand?

**Comment by dannylamb** [comment]
@ruebot oh, you did already.  Thanks.

**Comment by dannylamb** [comment]
@ruebot @whikloj We're green on before and after the revert.  This should be good to go now.

**Pull request merged**